File: AtlasWeb/Core/Models/LogisticDashboardViewModel.cs
using System;
using System.Collections.Generic;
using TTDataDictionary;
using TTObjectClasses;

namespace Core.Models
{
    public class LogisticDashboardViewModel
    {
        public bool hasRoleDashboardMaterialStatus { get; set; }
        public bool hasRoleDashboardCostAction { get; set; }
        public bool hasRoleDashboardMinMax { get; set; }
        public bool hasRoleSubStoreExpDateUpdate { get; set; }
        public bool hasRoleSubStoreWaitingWorks {get;set;}

        public bool hasRoleOpenCloseTerm { get; set; }

        public List<DashboardGridItemModel> dashboardGridItemModel
        {
            get;
            set;
        }

        public List<CostActionSelectBox> costActionSelectBox
        {
            get;
            set;
        }

        public List<ArchitectureInfo> architectureInfo
        {
            get;
            set;
        }

        public object Material
        {
            get;
            set;
        }
        public object MinMaxMaterial
        {
            get;
            set;
        }

        public List<MaterialDetail> materialDetail
        {
            get;
            set;
        }

        public List<StockCikis> stockCikis
        {
            get;
            set;
        }

        public List<StockGiris> stockGiris
        {
            get;
            set;
        }

        public List<StockActionWorkListDashboardItemModel> stockactionlist
        {
            get;
            set;
        }

        public List<CostActionAccountingTerm> costActionAccountingTerm
        {
            get;
            set;
        }

        public bool IsMainStoreFlag
        {
            get;
            set;
        }

        public List<MaterialInheldMaxMin> MaterialInheldMaxMin
        {
            get;
            set;
        }

        public List<MinMaxCalcTypeEnum> MinMaxCalcTypes
        {
            get;
            set;
        }

        public MinMaxCalcTypeEnum selectedMinMaxCalc
        {
            get;
            set;
        }

        public List<MinMaxCalculetedItem> MinMaxCalculetedItems
        {
            get;
            set;
        }

        public List<BudgetTypeSource> budgetTypeSources
        {
            get;
            set;
        }

        public BudgetTypeSource defaultBudgetType
        {
            get;
            set;
        }

        public CostActionAccountingTerm activeCostActionAccountingTerm
        {
            get;
            set;
        }
    }





    public class StockActionWorkListDashboardItemModel
    {
        public string ObjectID
        {
            get;
            set;
        }

        public Int32 StockActionID
        {
            get;
            set;
        }

        public TTObjectClasses.TransactionTypeEnum StockActionType
        {
            get;
            set;
        }

        public string DestinationStore
        {
            get;
            set;
        }

        public string StockactionName
        {
            get;
            set;
        }

        public string PatientName
        {
            get;
            set;
        }

        public DateTime TransactionDate
        {
            get;
            set;
        }

        public string StateName
        {
            get;
            set;
        }

        public string StateFormName
        {
            get;
            set;
        }

        public double Amount
        {
            get;
            set;
        }
    }

    public class DashboardGridItemModel
    {
        public string ObjectID
        {
            get;
            set;
        }

        public string Material
        {
            get;
            set;
        }

        public string MaterialName
        {
            get;
            set;
        }

        public Double AvarageUnitePrice
        {
            get;
            set;
        }

        public Double DifAvarageUnitePrice
        {
            get;
            set;
        }

        public Double MaterialPrice
        {
            get;
            set;
        }

        public Double PreviousMonthPrice
        {
            get;
            set;
        }

        public Double PreviousMothInheld
        {
            get;
            set;
        }

        public Double ProfitAndLoss
        {
            get;
            set;
        }

        public Double TotalAmount
        {
            get;
            set;
        }

        public Double TotalOutAmunt
        {
            get;
            set;
        }

        public Double TotalPrice
        {
            get;
            set;
        }

        public Double TransferredAmount
        {
            get;
            set;
        }
    }

    public class CostActionSelectBox
    {
        public string Objetid
        {
            get;
            set;
        }

        public string costActionDesciption
        {
            get;
            set;
        }
    }

    public class BudgetTypeSource
    {
        public string objectID
        {
            get;
            set;
        }

        public string name
        {
            get;
            set;
        }
    }

    public class CostActionAccountingTerm
    {
        public string ObjId
        {
            get;
            set;
        }

        public string Desciption
        {
            get;
            set;
        }

        public DateTime StartDate
        {
            get;
            set;
        }

        public AccountingTermStatusEnum? Status
        {
            get;
            set;
        }
    }

    public class ArchitectureInfo
    {
        public string yearKey
        {
            get;
            set;
        }

        public string year
        {
            get;
            set;
        }

        public Double cluster
        {
            get;
            set;
        }

        public string ObjId
        {
            get;
            set;
        }

        public string MaterialName
        {
            get;
            set;
        }
    }

    public class MaterialDetail
    {
        public string ObjectID
        {
            get;
            set;
        }

        public Int32 Inheld
        {
            get;
            set;
        }

        public Double totalIn
        {
            get;
            set;
        }

        public Double totalOut
        {
            get;
            set;
        }

        public string totalInPrice
        {
            get;
            set;
        }

        public string totalOutPrice
        {
            get;
            set;
        }

        public string materialName
        {
            get;
            set;
        }

        public string barcode
        {
            get;
            set;
        }

        public string distibutiontype
        {
            get;
            set;
        }

        public string materialClasses
        {
            get;
            set;
        }

        public byte[] materialPicture
        {
            get;
            set;
        }

        public string materialMKYSCod
        {
            get;
            set;
        }

        public string storageConditions
        {
            get;
            set;
        }

        public string estimatedTimeOfCheck
        {
            get;
            set;
        }

        public Double StockInheldMaxLevel
        {
            get;
            set;
        }

        public Double StockInheldMinLevel
        {
            get;
            set;
        }

        public List<StockDataInformation> StockDataInformation
        {
            get;
            set;
        }

        public List<MaterialInheldMaxMin> MaterialInheldMaxMin
        {
            get;
            set;
        }

        public List<StockDataLevelType> StockDataLevelType
        {
            get;
            set;
        }

        public List<StockBudgetType> StockBudgetType
        {
            get;
            set;
        }
    }

    public class MaterialInheldMaxMin
    {
        public double maxInheld
        {
            get;
            set;
        }

        public double minInheld
        {
            get;
            set;
        }

        public double inheld
        {
            get;
            set;
        }
    }

    public class StockBudgetType
    {
        public string BudgetType
        {
            get;
            set;
        }

        public double Amount
        {
            get;
            set;
        }
    }

    public class StockDataInformation
    {
        public string LotNo
        {
            get;
            set;
        }

        public DateTime ExpirationDate
        {
            get;
            set;
        }

        public Currency ResAmount
        {
            get;
            set;
        }
    }

    public class StockDataLevelType
    {
        public TTObjectClasses.StockLevelTypeEnum StockLevelType
        {
            get;
            set;
        }

        public double Amount
        {
            get;
            set;
        }
    }

    public class StockGiris
    {
        public string Description
        {
            get;
            set;
        }

        public double Amount
        {
            get;
            set;
        }

        public List<StockGirisDetay> stockGirisDetayList
        {
            get;
            set;
        }
    }

    public class StockCikis
    {
        public string Description
        {
            get;
            set;
        }

        public double Amount
        {
            get;
            set;
        }

        public List<StockCikisDetay> stockCikisDetayList
        {
            get;
            set;
        }
    }

    public class StockCikisDetay
    {
        public string description
        {
            get;
            set;
        }

        public double amount
        {
            get;
            set;
        }
    }

    public class StockGirisDetay
    {
        public string description
        {
            get;
            set;
        }

        public double amount
        {
            get;
            set;
        }
    }

    public class MinMaxCalculetedItem
    {
        public string StockObjectID
        {
            get;
            set;
        }

        public string MaterialName
        {
            get;
            set;
        }

        public string DistributionTypeName
        {
            get;
            set;
        }

        public double Inheld
        {
            get;
            set;
        }

        public double OutAmount
        {
            get;
            set;
        }

        public double MinLevel
        {
            get;
            set;
        }
        public double CriticalLevel { get; set; }
        public double MaxLevel
        {
            get;
            set;
        }

        public double CalcMinLevel
        {
            get;
            set;
        }

        public double CalcMaxLevel
        {
            get;
            set;
        }
        public double CalcCriticalLevel { get; set; }
    }


}

File: AtlasWeb/Core/wwwroot/app/Logistic/Models/LogisticDashboardViewModel.ts
import { BaseModel } from 'Fw/Models/BaseModel';
import { TransactionTypeEnum } from 'NebulaClient/Model/AtlasClientModel';
import { StockLevelTypeEnum } from 'NebulaClient/Model/AtlasClientModel';
import { MinMaxCalcTypeEnum } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from 'app/NebulaClient/Mscorlib/Guid';



export class LogisticDashboardViewModel extends BaseModel {

    dashboardGridItemModel: Array<DashboardGridItemModel>;
    costActionSelectBox: Array<CostActionSelectBox>;
    architectureInfo: Array<ArchitectureInfo>;
    Store: string;
    Material: string;
    MinMaxMaterial: string;
    materialDetail: Array<MaterialDetail>;
    stockCikis: Array<StockCikis>;
    stockGiris: Array<StockGiris>;
    stockactionlist: Array<StockActionWorkListDashboardItemModel> = new Array<StockActionWorkListDashboardItemModel>();
    StockDataInformation: Array<StockDataInformation>;
    costActionAccountingTerm: Array<CostActionAccountingTerm>;
    IsMainStoreFlag: boolean;
    MaterialInheldMaxMin: Array<MaterialInheldMaxMin>;
    MinMaxCalcTypes: Array<MinMaxCalcTypeEnum> = new Array<MinMaxCalcTypeEnum>();
    selectedMinMaxCalc: MinMaxCalcTypeEnum;
    MinMaxCalculetedItems: Array<MinMaxCalculetedItem>;

    public hasRoleDashboardMaterialStatus: boolean;
    public hasRoleDashboardCostAction: boolean;
    public hasRoleDashboardMinMax: boolean;
    public hasRoleSubStoreExpDateUpdate: boolean;
    public hasRoleSubStoreWaitingWorks: boolean;
    public hasRoleOpenCloseTerm: boolean;
    public budgetTypeSources: Array<BudgetTypeSource>;
    public defaultBudgetType: BudgetTypeSource;
    public activeCostActionAccountingTerm: CostActionAccountingTerm;
}

export class DashboardGridItemModel {
    public ObjectID: string;
    public Material: string;
    public MaterialName: string;
    public AvarageUnitePrice: number;
    public DifAvarageUnitePrice: number;
    public MaterialPrice: number;
    public PreviousMonthPrice: number;
    public PreviousMothInheld: number;
    public ProfitAndLoss: number;
    public TotalAmount: number;
    public TotalOutAmunt: number;
    public TotalPrice: number;
    public TransferredAmount: number;

}

export class CostActionAccountingTerm {
    public ObjId: string;
    public Desciption: string;
    @Type(() => Date)
    public StartDate: Date;
}

export class CostActionSelectBox {
    public Objetid: string;
    public costActionDesciption: any;
}

export class BudgetTypeSource {
    public objectID: string;
    public name: string;
}

export class ArchitectureInfo {
    public yearKey: string;
    public year: string;
    public cluster: number;
    public ObjId: string;
    public MaterialName: string;
}

export class DashboardListItem {
    public ObjectDefName: string;
    public ObjectDefId: string;
}

export class QueryInputDVO {
    @Type(() => Date)
    public StartDate: Date;
    @Type(() => Date)
    public EndDate: Date;
    public ObjId: string;
    public Store: string;
    @Type(() => Date)
    public Months: Date;
    public selectMonth: boolean;
    public Material: string;
    public accountingTermObjId: string;
}

export class SubStoreWorksInputDVO {
    @Type(() => Date)
    public StartDate: Date;
    @Type(() => Date)
    public EndDate: Date;
    public ObjectID: string;
    public Store: string;
    public AllReport: Number;
}

export class SubStoreWorksOutputDVO {
    public StoreName: string;
    public StockActionID: number;
    public DestinationStore: string;
    @Type(() => Date)
    public WorklistDate: Date;
    public StateName: string;
    public ObjectName: string;
    @Type(() => Guid)
    public ObjectID: Guid;
}
export class MaterialDetail {
    public ObjectID: string;
    public Inheld: number;
    public totalIn: number;
    public totalOut: number;
    public totalInPrice: string;
    public totalOutPrice: string;
    public materialName: string;
    public barcode: string;
    public distibutiontype: string;
    public materialClasses: string;
    public materialMKYSCod: string;
    public storageConditions: string;
    public estimatedTimeOfCheck: string;
    public materialPicture: number[];
    public StockInheldMaxLevel: number;
    public StockInheldMinLevel: number;
    public StockDataInformation: Array<StockDataInformation>;
    public StockBudgetType: Array<StockBudgetType>;
    public StockDataLevelType: Array<StockDataLevelType>;

    public MaterialInheldMaxMin: Array<MaterialInheldMaxMin>;
}
export class StockBudgetType {
    public BudgetType: string;
    public Amount: number;
}
export class MaterialInheldMaxMin {
    public maxInheld: number;
    public minInheld: number;
    public inheld: number;
}
export class StockDataInformation {
    public LotNo: string;
    @Type(() => Date)
    public ExpirationDate: Date;
    public ResAmount: number;
}
export class StockDataLevelType {
    public StockLevelType: StockLevelTypeEnum;
    public Amount: number;
}


export class StockGiris {
    public Description: string;
    public Amount: number;
    public stockGirisDetayList: Array<StockGirisDetay>;
}

export class StockCikis {
    public Description: string;
    public Amount: number;
    public stockCikisDetayList: Array<StockCikisDetay>;
}

export class StockCikisDetay {
    public description: string;
    public amount: number;
}

export class StockGirisDetay {
    public description: string;
    public amount: number;
}


export class StockActionWorkListDashboardItemModel {

    public ObjectID: string;
    public StockActionID: number;
    public StockActionType: TransactionTypeEnum;
    public DestinationStore: string;
    public StockactionName: string;
    public PatientName: string;
    @Type(() => Date)
    public TransactionDate: Date;
    public StateName: string;
    public StateFormName: string;
    public Amount: number;
}

export class DynamicComponentInfoDVO {
    public ComponentName: string;
    public ModuleName: string;
    public ModulePath: string;
    public objectID: string;
}

//VADEMECUM INSTANCES START POINT
export class QueryVademecumInteractionDVO {
    public episodeID: string;
    public prdList: Array<Product>;
    public diseaseList: Array<Disease>;
    public icdCodeList: Array<ICDCode>;
    public symptomList: Array<Symptom>;
    public nutritionList: Array<Nutrition>;
    public patientCharacteristics: PatientCharacteristics;
}

export class QueryVademecumInteractionDVONew {
    public episodeID: string;
    public prdList: Array<Guid>;
    public diseaseList: Array<Disease>;
    public icdCodeList: Array<ICDCode>;
    public symptomList: Array<Symptom>;
    public nutritionList: Array<Nutrition>;
    public patientCharacteristics: PatientCharacteristics;
}
export class VademecumProductDVO {
    public id: string;
}
export class Disease {
    public id: number;
}
export class ICDCode {
    public name: string;
}
export class Nutrition {
    public id: number;
}
export class PatientCharacteristics {
    public weight: number;
    public height: number;
    @Type(() => Date)
    public birthdate: Date;
    @Type(() => Date)
    public pregnancyDate: Date;
    public gender: string;
    public smokingHabit: boolean;
    public drugHabit: boolean;
    public pregnancyProbability: boolean;
    public breastFeeding: boolean;
    public contactLensUsage: boolean;
    public saltFreeDiet: boolean;
    public electroshockTherapy: boolean;
    public postpartumPeriod: boolean;
    public bloodDonation: boolean;
    public pregnant: boolean;
}
export class Product {
    public id: number;
}
export class Symptom {
    public id: number;
}
//VADEMECUM INSTANCES END POINT

export class MinMaxCalculetedItem {
    public StockObjectID: string;
    public MaterialName: string;
    public DistributionTypeName: string;
    public Inheld: number;
    public OutAmount: number;
    public MinLevel: number;
    public CriticalLevel: number;
    public MaxLevel: number;
    public CalcMinLevel: number;
    public CalcMaxLevel: number;
    public CalcCriticalLevel: number;
}
export class CalcMinMaxValueDVO {
    public id: string;
    public selectedType: MinMaxCalcTypeEnum;
    public StartDate: Date;
    public EndDate: Date;
    public materialObjectID: Array<string> = new Array<string>();
    public AllProducts: boolean;
    public StockHasInheld: boolean;
}
export class UpdateMinMaxValueDVO {
    public MinMaxCalculetedItems: Array<MinMaxCalculetedItem>;
}

export class InStockTransactionInput {
    public store: string;
    public materials: Array<string> = new Array<string>();
    public accountingTermObjId: string;
    public budgetTypeObjId: string;
    public isZeroAmount: boolean;
}

export class InStockTransactionDVO {
    public materialName: string;
    public restAmount: number;
    public unitPrice: number;
    public stockNo: string;
    @Type(() => Date)
    public transactionDate: Date;
    public inAmount: number;
    public outAmount: number;
    @Type(() => Date)
    public expirationDate: Date;
    public minLevel: number;
    public maxLevel: number;
    public barcode: string;
    public MKYSTrxID: number;
    public stockTransactionObjectID: Guid;
    public trxDescription: string;
    public destinationDescription: string;
    public auctionNo: string;
    public stockActionId: number;
    public tifNo: string;
    public stockTransactionListObjectID: Array<Guid>;

}

export class OutStockTransactionInput {
    public inStockTransactionID: string;
    public accountingTermObjId: string;
    public inStockTransactionListID: Array<string>;
}

export class OutStockTransactionDVO {
    @Type(() => Date)
    public transactionDate: Date;
    public trxType: string;
    public trxDescription: string;
    public storeName: string;
    public documentLogID: number;
    public stockActionID: number;
    public amount: number;
    @Type(() => Date)
    public expirationDate: Date;
    public stockTransactionObjectID: Guid;
}

export class SubStoreExpDateUpdate_Input {
    public materials: Array<string> = new Array<string>();
    public store: string;
    public beforeDay: number;
}

export class SubStoreExpDateUpdate_Output {
    public transactionObjectID: Guid;
    public materialName: string;
    public inheld: number;
    @Type(() => Date)
    public expDate: Date;
    public dayDiff: number;
    @Type(() => Date)
    public newExpDate: Date;
}
export class TransactionChartInputDTO {
    public StockInList = new Array<StockGiris>();
    public StockOutList = new Array<StockCikis>();
} 